TsJournalTrello.gs

```diff
@@ -0,0 +1,361 @@
+      .getSheetByName('Organisations')
+      .getDataRange()
+  
+  var data = orgData.getValues()
+  var timesheetUrl = null
+  var trelloBoardUrl = null
+  var trelloBoardId = null
+
+  // Get the organisation name from the journal name, text before '- journal' 
+  var str = ' - Journal'
+  var CHAR_TO_END_ = str.length
+  var orgName = DocumentApp.getActiveDocument().getName().slice(0, -CHAR_TO_END_)
+  var orgFound = false
+    
+  //Loop through the rows of data
+  orgFound = data.some(function(row)  {
+      
+    //check if the company name matches the orgName
+    var orgNameSearch = row[ORG_NAME_COL_]
+      
+    if (orgNameSearch === orgName) {
+    
+      //Get the timesheet and Trello Board Url
+      timesheetUrl = row[ORG_TS_COL_]
+      trelloBoardUrl = row[TRELLO_COL_]
+      return true
+      
+      }
+      
+  })   
+    
+  if (!orgFound) {
+    
+    Log_.warning('Organisation Name: ' + orgName + ' not found in Org Spreadsheet')
+    return
+      
+  } else {
+    
+    Log_.info('Organisation Name: ' + orgName + ' found in Org Spreadsheet')
+      
+  }
+   
+    
+  //Check the Trello Board Exists  
+  if (trelloBoardUrl === '') {
+    
+    Log_.warning('No Trello Board URL found in Org Spreadsheet')
+    return
+      
+  } else {
+      
+    //Get the Trello Board Id from the wedsite json return  
+    trelloBoardId = getTrelloBoardId_(trelloBoardUrl)
+    
+    if (trelloBoardId == null) {
+      Log_.warning('No new trello board Id found for ' + trelloBoardUrl)
+      return
+    }    
+    Log_.info('Trello Card Id: ' + trelloBoardId)
+        
+  } //if (trelloBoardUrl === '')
+      
+  //Open the Journal, loop through the paragraphs until the Heading 1 is found
+  //The next paragraph is the Trello Card Name, check it hasnt already been processed.
+  //If it hasnt, change the style to Heading 2 and add a Hyperlink to the Trello Card 
+  var journal = DocumentApp.openById(DocId)
+  var DocUrl = journal.getUrl()
+  var npBookmark = null
+ 
+  if (journal === null) { 
+  
+    Log_.warning('Invalid Journal ID' + DocId)
+   
+    return
+  } else {
+   
+    //Process the journal, find the next tello card task, hyperlink the heading
+    var [trelloCardTitle, npBookmarkId] = processJournal_(journal, trelloBoardId)
+          
+    if (trelloCardTitle == null) {
+      Log_.warning('Either no trello card title found or trello card found on trello board')
+      return
+    } else if (npBookmarkId == null) {
+      Log_.warning('No Bookmark Found')    
+      return
+    } else {
+      //Add the link to the timesheet
+      processTimesheet_(timesheetUrl, npBookmarkId, DocUrl, trelloCardTitle)    
+    }
+  }
+                
+  return SUCCESS  
+}// linkHeaders() 
+
+
+function processJournal_(journal, trelloBoardId) {
+          
+  var paragraphs = journal.getBody().getParagraphs()
+  var ProcessJournalReturn = null
+  var Heading1Paragraphs = []
+  
+  for(var i in paragraphs) { 
```

Comment:
[for/in is for processing objects not arrays](https://google.github.io/styleguide/jsguide.html#features-for-loops).

forEach is faster and more appropriate:

paragraphs.forEach(function(paragraph, paragraphIndex) { ... })

Reply:
Ok, yes will change